### DISCUSSION TITLE
How to extract image with same size and content as pdf displayed?

### DISCUSSION THREAD

**Discussion by weikenxq** [pymupdf/PyMuPDF#1825] [Q&A] [answered]
Hi
When i try to extract image from pdf, i found that the size of image not the same as pdf displayed. It is possible to  extract image with same size and content as pdf displayed?

below is the pdf file
[image1.pdf](https://github.com/pymupdf/PyMuPDF/files/9143054/image1.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
> It is possible to extract image with same size as pdf displayed?

No, because you are extracting the original image file.
The only thing you can do is to **_afterwards_** modify the extracted image to a size of your liking.
Do this by applying a scaling matrix to the pixmap of the extracted image, or make a PIL Image and then execute its `.thumbnail()` method.

**Reply by weikenxq** [reply]
thanks for your fast reply.
but how can i get the image area displayed on the pdf?
i have try get the bbox of the image, but the size of the bbox not the same as the pdf displayed.

**Reply by JorjMcKie** [reply]
For example like this:
```python
for item in page.get_images():
    xref = item[0]  # xref of image
    bbox, matrix = page.get_image_rects(xref, transform=True)[0]
    # bbox is the rectangle shown on the page
    # matrix is the transformation applied to Rect(0, 0, width, height) of original image
    # please consult documentation!
```

**Reply by weikenxq** [reply]
Thanks for reply again. 
I have read the documentation about bbox and matrix, but i do not know how to do it.  How can i to apply the bbox and matrix to the origin image? Is there some tools may use，or could you please give me an example?

**Reply by JorjMcKie** [reply]
Okay, that's good, so you know what I am talking about here:
1. The bbox on the page for the original image is computed by `bbox = imgrect * shrink * transform`.
2. So, if no rotation of the original image was involved, the matrix `mat = shrink * transform` contains the x- and y-scaling factors in its property `mat.a`, resp. `mat.d`
3. Then do `imgpix = fitz.Pixmap(doc.extract_image(xref)["image"])`.
4. Now make a pixmap copy by scaling width and height of imgrect: `bboxpix = fitz.Pixmap(imgpix, width / mat.a, height / mat.d)`.
5. Save `bboxpix.save("scaled.png")`.

**Reply by JorjMcKie** [reply]
You can of course use arbitrary values for the new width and height you want for `bboxpix`. An appropriate scaling in each dimension will automatically happen.

**Reply by JorjMcKie** [reply]
And another comment:
If you want to evade all this trouble, you can also make a pixmap of the image on the page directly:
Simply compute the bbox, then do `bboxpix = page.get_pixmap(clip=bbox)`. Done.
